wire-ios/Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift

```diff
@@ -73,81 +98,95 @@ final class LocationSelectionViewController: UIViewController {
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        if !userLocationAuthorized { mapView.restoreLocation(animated: true) }
-        locationManager.requestWhenInUseAuthorization()
-        updateUserLocation()
-
+        if !userLocationAuthorized { mapViewController.mapView.restoreLocation(animated: true) }
+        appLocationManager.requestLocationAuthorization()
         endEditing()
     }
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        locationManager.stopUpdatingHeading()
-        mapView.storeLocation()
+        appLocationManager.stopUpdatingLocation()
+        mapViewController.mapView.storeLocation()
     }
 
-    fileprivate func configureViews() {
+    // MARK: - Configuration
+
+    private func configureViews() {
         addChild(sendViewController)
         sendViewController.didMove(toParent: self)
-        [mapView, sendViewController.view, toolBar, locationButton].forEach(view.addSubview)
-        locationButton.addTarget(self, action: #selector(locationButtonTapped), for: .touchUpInside)
 
-        mapView.isRotateEnabled = false
-        mapView.isPitchEnabled = false
+        view.addSubview(mapViewController.mapView)
+        view.addSubview(sendViewController.view)
+        view.addSubview(toolBar)
+        view.addSubview(locationButton)
+
+        let action = UIAction { [weak self] _ in
+            self?.locationButtonTapped()
+        }
+
+        locationButton.addAction(action, for: .touchUpInside)
+
         toolBar.configure(title: title ?? "", subtitle: nil, topAnchor: safeTopAnchor)
-        pointAnnotation.coordinate = mapView.centerCoordinate
+        pointAnnotation.coordinate = mapViewController.mapView.centerCoordinate
 
-        mapView.addSubview(annotationView)
-    }
+        mapViewController.mapView.addSubview(annotationView)
 
-    fileprivate func createConstraints() {
+    }
+    private func createConstraints() {
         guard let sendController = sendViewController.view else { return }
 
-        [mapView, sendController, annotationView, toolBar, locationButton].forEach {
+        [mapViewController.mapView, sendController, annotationView, toolBar, locationButton].forEach {
```

Comment:
The PR's description says:
> Created MapViewController class that encapsulates the MKMapView and its configurations.

So I am asking myself if it's enough to provide an encapsulate `UIView` and avoid to expose the type `MKMapView`, because this can be configured in any way and can skip the controller. One example from line 326:
`mapViewController.mapView.showsUserLocation = true`